libsolidity/ast/Types.cpp

```diff
@@ -536,8 +536,7 @@ bool FixedPointType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 bool FixedPointType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
 	return _convertTo.category() == category() ||
-		_convertTo.category() == Category::Integer ||
-		_convertTo.category() == Category::FixedBytes;
```

Comment:
I think all the changes done to `ast/Types.cpp` could be pulled into a separate PR and comprehensive test cases can be added to the `syntaxTests` directory.